flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/catalog/AbstractJdbcCatalog.java

```diff
@@ -116,6 +122,36 @@ public String getBaseUrl() {
 		return baseUrl;
 	}
 
+	// ------ Postgres default objects that shouldn't be exposed to users ------
+
+	/**
+	 * Retrieve the list of system schemas to ignore.
+	 */
+	public abstract Set<String> getBuiltinSchemas();
+
+	/**
+	 * Retrieve the list of system database to ignore.
+	 */
+	public abstract Set<String> getBuiltinDatabases();
+
+	// ------ retrieve PK constraint ------
+
+	protected Map.Entry<String, List<String>> getPrimaryKey(DatabaseMetaData metaData, PostgresTablePath pgPath) throws SQLException {
+		ResultSet rs = metaData.getPrimaryKeys(null,  pgPath.getPgSchemaName(), pgPath.getPgTableName());
+		Map.Entry<String, List<String>> ret = null;
+		while (rs.next()) {
+			String schema = rs.getString("table_schem");
+			String columnName = rs.getString("column_name");
+			String pkName = rs.getString("pk_name");
+			if (!getBuiltinSchemas().contains(schema)) {
+				if (ret == null) {
+					ret = new AbstractMap.SimpleEntry<>(pkName, new ArrayList<>());
+				}
+				ret.getValue().add(columnName);
```

Comment:
According to the Javadoc of `java.sql.DatabaseMetaData#getPrimaryKeys`, the returned primary key columns are ordered by COLUMN_NAME, not KEY_SEQ. We may need to sort them again based on the KEY_SEQ. I know that posgres return values ordered by the KEY_SEQ, but not all the dialects do like this.